library/platform_util.c

```diff
@@ -245,8 +247,11 @@ int mbedtls_platform_memequal( const void *buf1, const void *buf2, size_t num )
 
     /* Start from a random location and check the correct number of iterations */
     size_t i, flow_counter = 0;
-    size_t start_offset = (size_t) mbedtls_platform_random_in_range( (uint32_t) num );
-
+    size_t start_offset = 0;
+    if( num > 0 )
```

Comment:
It seems we moved the problematic "if" condition from random to memequal  and memset. Now we have the same problem( less critical than in random ) - we can skip the "if" and always start from zero offset.
Let's try to create dependency between flow_counter and num, as we did it in memcmp

Reply:
Moved the checks, so that now this if cannot be easily skipped, and if is (with num being non-zero) the flow control check will fail.